vpr/src/route/rr_graph_generation/rr_graph.cpp

```diff
@@ -2011,6 +2041,91 @@ static void build_rr_chan(RRGraphBuilder& rr_graph_builder,
     }
 }
 
+static void add_and_connect_non_3d_sg_links(RRGraphBuilder& rr_graph_builder,
+                                            const std::vector<t_bottleneck_link>& sg_links,
+                                            const std::vector<std::pair<RRNodeId, int>>& sg_node_indices,
+                                            const t_chan_details& chan_details_x,
+                                            const t_chan_details& chan_details_y,
+                                            size_t num_seg_types_x,
+                                            t_rr_edge_info_set& non_3d_sg_rr_edges_to_create) {
+    VTR_ASSERT(sg_links.size() == sg_node_indices.size());
+    const size_t num_links = sg_links.size();
+
+    for (size_t i = 0; i < num_links; i++) {
+
+        const t_bottleneck_link& link = sg_links[i];
+
+        int xlow, xhigh, ylow, yhigh;
+        Direction direction;
+        const t_physical_tile_loc& src_loc = link.gather_loc;
+        const t_physical_tile_loc& dst_loc = link.scatter_loc;
+
+        VTR_ASSERT_SAFE(src_loc.layer_num == dst_loc.layer_num);
+        const int layer = src_loc.layer_num;
+
```

Comment:
could this be refactored with another block of code (which I think was the spatial lookup insertion code) to have a helper once)?